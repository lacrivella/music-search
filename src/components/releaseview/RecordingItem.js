import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function RecordingItem({ recording, artistName }) {
  return (
    <Link to={`/song/${artistName}/${recording.title}`}>
      <section>
        <p>{recording.title}</p>
      </section>
    </Link>
  );
}

RecordingItem.propTypes = {
  recording: PropTypes.shape({
    title: PropTypes.string.isRequired
  }),
  artistName: PropTypes.string
};

export default RecordingItem;
