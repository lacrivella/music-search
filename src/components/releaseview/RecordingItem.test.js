import React from 'react';
import { shallow } from 'enzyme';
import RecordingItem from './RecordingItem';

describe('RecordingItem component', () => {
  it('renders App', () => {
    const wrapper = shallow(<RecordingItem recording={{ title: 'test' }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
