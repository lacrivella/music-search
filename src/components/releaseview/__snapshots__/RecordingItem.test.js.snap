// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecordingItem component renders App 1`] = `
<Link
  to="/song/undefined/test"
>
  <section>
    <p>
      test
    </p>
  </section>
</Link>
`;
